#!/usr/bin/env python
PACKAGE = "ark_controls"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("kpx",    double_t,    0, "kpx", 1000,  0, 5000)
gen.add("kix",    double_t,    0, "kix", 0,  0, 5000)
gen.add("kdx",    double_t,    0, "kdx", 0,  0, 5000)
gen.add("kpy",    double_t,    0, "kpy", 0,  0, 5000)
gen.add("kiy",    double_t,    0, "kiy", 0,  0, 5000)
gen.add("kdy",    double_t,    0, "kdy", 0,  0, 5000)
gen.add("accelmax_x",    double_t,    0, "accelmax_x", 0.25,  0, 5)
gen.add("accelmax_y",    double_t,    0, "accelmax_y", 0.25,  0, 5)
gen.add("deaccelmax_x",    double_t,    0, "deaccelmax_x", 0.25,  0, 5)
gen.add("deaccelmax_y",    double_t,    0, "deaccelmax_y", 0.25,  0, 5)
gen.add("vmax_x",    double_t,    0, "vmax_x", 0.5,  0, 5)
gen.add("vmax_y",    double_t,    0, "vmax_y", 2,  0, 5)
gen.add("inr",    double_t,    0, "inr", 0.02,  0, 10)


'''size_enum = gen.enum([ gen.const("Small",      int_t, 0, "A small constant"),
                       gen.const("Medium",     int_t, 1, "A medium constant"),
                       gen.const("Large",      int_t, 2, "A large constant"),
                       gen.const("ExtraLarge", int_t, 3, "An extra large constant")],
                     "An enum to set size")'''

'''gen.add("size", int_t, 0, "A size parameter which is edited via an enum", 1, 0, 3, edit_method=size_enum)'''

exit(gen.generate(PACKAGE, "ark_controls", "Variables"))
